### PR TITLE
RTC-13528 Create dockerized LoadTest

### PR DESCRIPTION
**Pull request by reddvl1980**

### CODE REVIEW

File: bridge/Bridge.h [hunk 1]
```diff
@@ -43,6 +43,8 @@ class Engine;
 class MixerManager;
 class ApiRequestHandler;
 
+std::vector<transport::SocketAddress> gatherInterfaces(const config::Config& config);
```
Comment (RicardoMDomingues) [inline bridge/Bridge.h:46]: Perhaps it make sense to move this other place.

File: docker/ubuntu-focal-loadtest/startup.sh [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    gcloud compute instances delete -q "$INSTANCE_ID" --zone "$GCE_ZONE"
+fi
+}
+
+trap cleanup EXIT
+
+if [ -z "$INSTANCE_ID" ]
+then
+      echo "\$INSTANCE_ID env var should be set!"
+      exit
```
Comment (RicardoMDomingues) [inline docker/ubuntu-focal-loadtest/startup.sh:16]: Sounds this should exit with an error code. And perhaps is not "should" but "must"

File: docker/ubuntu-focal-loadtest/startup.sh [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+fi
+
+if [ -z "$GCE_BUCKET_NAME" ]
+then
+      echo "\$GCE_BUCKET_NAME env var should be set!"
+      exit
```
Comment (RicardoMDomingues) [inline docker/ubuntu-focal-loadtest/startup.sh:22]: ditto

File: docker/ubuntu-focal-loadtest/startup.sh [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+fi
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+if [ "$#" -ne 9 ]; then
+    echo "Usage startup.sh <initiator|guest> <SMB IP> <SMB Port> <num clients per instance> <num instances> <gtest filter> <joinRampUpTime> <maxJoinRampUpTime> <duration>. Eg. for 1000 clients: startup.sh initiator 127.0.0.1 8080 250 4 RealTimeTest.DISABLED_smbMegaHoot 30"
+    exit
```
Comment (RicardoMDomingues) [inline docker/ubuntu-focal-loadtest/startup.sh:29]: Bad arguments, exit must exit with error code

File: docker/ubuntu-focal-loadtest/startup.sh [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+fi
+
+if [ "$1" != "initiator" ] && [ "$1" != "guest" ]; then
+    echo "First param should be <initiator|guest>"
+    exit
```
Comment (RicardoMDomingues) [inline docker/ubuntu-focal-loadtest/startup.sh:34]: ditto

File: docker/ubuntu-focal-loadtest/startup.sh [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+if [ "$1" == "initiator" ]; then
+
+    # Create conference and obtain CONFERENCE_ID.
+    source $SCRIPT_DIR/create_conference.sh $2 $3 2>&1 | tee -a $log_file
+
+    if [ -f "conference_id.json" ]; then
+        CONFERENCE_ID=$(jq '.id' conference_id.json | head -n 1 | tr -d '"')
+        echo "Conference ID: $CONFERENCE_ID" 2>&1 | tee -a $log_file
+    else
+        exit
```
Comment (RicardoMDomingues) [inline docker/ubuntu-focal-loadtest/startup.sh:50]: Is it an error?

File: docker/ubuntu-focal-loadtest/startup.sh [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    fi
+
+    if [ -z "${CONFERENCE_ID}" ]
+    then
+      echo "create_conference.sh failed! Exiting..." 2>&1 | tee -a $log_file
+      exit
```
Comment (RicardoMDomingues) [inline docker/ubuntu-focal-loadtest/startup.sh:56]: ditto

File: docker/ubuntu-focal-loadtest/startup.sh [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        attempt=$(( attempt - 1 ))
+        echo "Trying to get config... attempt: $attempt" 2>&1 | tee -a $log_file
+        sleep 1
+    done
+    date -u 2>&1 | tee -a $log_file
+fi
+
+if [ ! -f load_test_config.json ]; then
+    echo "Failed to receive config file load_test_config.json. Exiting..."  2>&1 | tee -a $log_file
+    exit
```
Comment (RicardoMDomingues) [inline docker/ubuntu-focal-loadtest/startup.sh:90]: ditto

File: docker/ubuntu-focal-loadtest/wait_for_others.sh [hunk 9]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+set -e
+
+if [ -z "$INSTANCE_ID" ]
+then
+      echo "\$INSTANCE_ID env var should be set!"
+      exit
```
Comment (RicardoMDomingues) [inline docker/ubuntu-focal-loadtest/wait_for_others.sh:7]: ditto

File: docker/ubuntu-focal-loadtest/wait_for_others.sh [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+fi
+
+if [ -z "$GCE_BUCKET_NAME" ]
+then
+      echo "\$GCE_BUCKET_NAME env var should be set!"
+      exit
```
Comment (RicardoMDomingues) [inline docker/ubuntu-focal-loadtest/wait_for_others.sh:13]: ditto

File: docker/ubuntu-focal-loadtest/wait_for_others.sh [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+fi
+
+if [ "$#" -ne 2 ]; then
+    echo "Usage wait_for_others.sh <num instances> <timeout in seconds>. Eg. wait_for_others.sh 4 30"
+    exit
```
Comment (RicardoMDomingues) [inline docker/ubuntu-focal-loadtest/wait_for_others.sh:18]: ditto

File: test/integration/RealTimeTest.cpp [hunk 12]
```diff
@@ -171,49 +190,83 @@ bool RealTimeTest::isActiveTalker(const std::vector<api::ConferenceEndpoint>& en
...
+    uint16_t duration = 60;
+    uint32_t rampup = 0;
+    uint32_t max_rampup = 0;
+    utils::MersienneRandom<uint32_t> randGen;
+
+    if (_configInitialized)
+    {
+        numClients = _config->numClients;
+        duration = _config->duration;
+        // rampup = _config->rampup;
```
Comment (RicardoMDomingues) [inline test/integration/RealTimeTest.cpp:205]: Remove this line

File: docker/ubuntu-focal-loadtest/startup.sh (hunk 2 above)
Comment (reddvl1980) [reply docker/ubuntu-focal-loadtest/startup.sh:16]: Fixed,

File: bridge/Bridge.h (hunk 1 above)
Comment (reddvl1980) [reply bridge/Bridge.h:46]: Moved to new files utils\GatherInterfaces.h/.cpp

File: docker/ubuntu-focal-loadtest/startup.sh (hunk 6 above)
Comment (reddvl1980) [reply docker/ubuntu-focal-loadtest/startup.sh:50]: all exists nor return non-zero

**Review by RicardoMDomingues — COMMENTED** [review]
LGTM

**Review by RicardoMDomingues — APPROVED** [review]

**Pull request merged**